pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -352,6 +362,19 @@ func (ku *Upgrader) upgradeAgentPools(ctx context.Context) error {
 			}
 			ku.logger.Infof("Upgrading Agent VM: %s, pool name: %s", vm.name, *agentPool.Name)
 
+			// copy custom properties from old node to new node if the flag in AgentPoolProfile is set.
+			if agentPoolProfile != nil && agentPoolProfile.PreserveNodesPropertiesInUpgrading != nil && *agentPoolProfile.PreserveNodesPropertiesInUpgrading {
+				if len(newCreatedVMs) > 0 {
+					newNodeName := newCreatedVMs[0]
+					newCreatedVMs = newCreatedVMs[1:]
+					ku.logger.Infof("Copying custom annotations, labels, taints from old node %s to new node %s...", vm.name, newNodeName)
```

Comment:
isn't the new created VM name going to be the same as the old VM? Since we're replacing the VM

Reply:
no. upgrading add a new node and delete an old node one by one. The newly created VM doesn't have the same name as old name.  That's the reason that Azure #549 exists.

Reply:
agreed that the new VM is different and doesn't have the tags/labels the old one had but isn't the VM name the same?

Reply:
Checked an upgrade test, the VM name does stay the same I believe... Eg. `k8s-master-23770275-0` before upgrade, during upgrade a new `k8s-master-23770275-0` VM is created. Did you run an upgrade from your branch with tags to test this change? What does this log line print?

Reply:
VM name is not the same.